test/merge_test.dart

```diff
@@ -0,0 +1,102 @@
+import 'dart:async';
+
+import 'package:test/test.dart';
+
+import 'package:stream_transform/stream_transform.dart';
+
+import 'util/matchers.dart';
+
+void main() {
+  group('merge', () {
+    test('includes all values', () async {
+      var first = new Stream.fromIterable([1, 2, 3]);
+      var second = new Stream.fromIterable([4, 5, 6]);
+      var allValues = await first.transform(merge(second)).toList();
+      expect(allValues, containsAllInOrder([1, 2, 3]));
+      expect(allValues, containsAllInOrder([4, 5, 6]));
+    });
+
+    test('cancels both sources', () async {
+      var firstCanceled = false;
+      var first = new StreamController()
+        ..onCancel = () {
+          firstCanceled = true;
+        };
+      var secondCanceled = false;
+      var second = new StreamController()
+        ..onCancel = () {
+          secondCanceled = true;
+        };
+      var subscription =
+          first.stream.transform(merge(second.stream)).listen((_) {});
+      await subscription.cancel();
+      expect(firstCanceled, true);
+      expect(secondCanceled, true);
+    });
+  });
+
+  group('mergeAll', () {
+    test('includes all values', () async {
+      var first = new Stream.fromIterable([1, 2, 3]);
```

Comment:
I could add a 'does not add extra values' test... I would be surprised if we found a way to introduce that bug but it won't hurt to test :)